src/test/suite/file-controller.test.ts

```diff
@@ -34,3 +35,91 @@ suite('Extension Test Suite', () => {
         assert.deepEqual(paths, ["/hoge/test/bar", "/hoge/test", "/hoge/test/foo/bar", "/hoge/test/foo"]);
     });
 });
+
+
+// HACK: the compiled js its inside `out` and not `src`
+// although there might be better ways, it works for now
```

Comment:
Just comment: Maybe the best solution is mocking some input/output methods.
But this project is not introduced testing on CI.
So, I think your solution is OK at this time.

Reply:
Yes, this would be better, but I didn't want to overthink it too much.

I usually have a demo folder in my root. Then I can add the `${workspacefolder}/demo` path to the arguments of the `launch.json`, and vscode would launch the extension directly inside that folder.

From there, I can demo the extension commands and place the test files inside.

It might be a simpler/cleaner solution for this case.